src/basic/stat-util.c

```diff
@@ -163,6 +163,24 @@ int null_or_empty_fd(int fd) {
         return null_or_empty(&st);
 }
 
+int fd_is_read_only_fs(int fd) {
+        struct statvfs st;
+
+        assert(fd >= 0);
+
+        if (fstatvfs(fd, &st) < 0)
+                return -errno;
+
+        if (st.f_flag & ST_RDONLY)
+                return true;
+
+        /* See comments in path_is_read_only_fs(). */
+        if (faccessat(fd, ".", W_OK, 0) < 0 && errno == EROFS)
+                return true;
+
+        return false;
+}
```

Comment:
hmm, I think it would make sense to rework path_is_read_only_fs() on top of fd_is_read_only_fs() if we have that now. i.e. open the path with O_PATH, then issue fd_is_read_only_fs() on it, and that#s it.

Given we do two checks here, it's kinda nice to "pin" the inode inbetween,  instead of going through the path twice.